modin/pandas/test/test_series.py

```diff
@@ -4705,3 +4705,30 @@ def func(val):
     modin_series = modin_df["col"].apply(func)
 
     df_equals(modin_series, pandas_series)
+
+
+def test_apply_return_df():
+    modin_df = pd.Series(1).apply(lambda x: pandas.Series([x, x]))
+    pandas_df = pandas.Series(1).apply(lambda x: pandas.Series([x, x]))
+
+    df_equals(modin_df, pandas_df)
+
+    modin_df = pd.Series([2, 4, 6, 0, 1]).apply(lambda x: pandas.Series([x, x + 1]))
+    pandas_df = pandas.Series([2, 4, 6, 0, 1]).apply(
+        lambda x: pandas.Series([x, x + 1])
+    )
+
+    df_equals(modin_df, pandas_df)
+
+
+@pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
+def test_apply_return_df_multiple_partitions(data):
+    modin_series, pandas_series = create_test_series(data)
+    modin_df = modin_series.apply(
+        lambda x: pandas.Series([x, x + 1, x + 2], index=["a", "b", "c"])
+    )
+    pandas_df = pandas_series.apply(
+        lambda x: pandas.Series([x, x + 1, x + 2], index=["a", "b", "c"])
```

Comment:
Should we also test when there will be multiple horizontal partitions too? I.e. increase the number of columns to make sure things get block partitioned appropriately and don't become long axis chunks?

Reply:
+1, but we can just do a single test case where the function result spans multiple horizontal partitions.

Reply:
After testing it seems like what's happening is that we are getting long axis chunks. The number of partitions isn't changing after doing an apply that creates lots of columns.